api/src/middleware/assessmentsDummyRouter.ts

```diff
@@ -0,0 +1,291 @@
+    try {
+      insertedAssessmentSubmissionsRow = await insertToAssessmentSubmissions(
+        dummyAssessmentSubmissionData[0].assessment_id,
+        dummyAssessmentSubmissionData[0].principal_id,
+        dummyAssessmentSubmissionData[0].assessment_submission_state_id,
+        dummyAssessmentSubmissionData[0].score,
+        dummyAssessmentSubmissionData[0].opened_at,
+        dummyAssessmentSubmissionData[0].submitted_at,
+        dummyAssessmentSubmissionData[0].responses
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+
+    res.status(201).json(itemEnvelope(insertedAssessmentSubmissionsRow));
+  }
+);
+
+assessmentsDummyRouter.get(
+  '/makeFacilitator/:programId/:participantId',
+  async (req, res, next) => {
+    const { programId, participantId } = req.params;
+    // a role ID of 2 corresponds to a Facilitator
+    const roleId = 2;
+
+    const programIdParsed = Number(programId);
+    const participantIdParsed = Number(participantId);
+    if (
+      !Number.isInteger(programIdParsed) ||
+      programIdParsed < 1 ||
+      !Number.isInteger(participantIdParsed) ||
+      participantIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${programIdParsed}""${participantIdParsed}"  is not a valid program id .`
+        )
+      );
+      next(
+        new BadRequestError(
+          `"${participantIdParsed}"  is not a valid participant id .`
+        )
+      );
+      return;
+    }
+
+    let insertedProgramFacilitatorsRow: ProgramParticipantsRow;
+    try {
+      insertedProgramFacilitatorsRow = await insertToProgramParticipants(
+        participantIdParsed,
+        programIdParsed,
+        roleId
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+    res.status(201).json(itemEnvelope(insertedProgramFacilitatorsRow));
+  }
+);
+interface AssessmentStartRow {
+  id: number;
+  assessment_id?: number;
+  principal_id?: number;
+}
+assessmentsDummyRouter.get(
+  '/startAssessment/:assessmentId/:participantId',
+  async (req, res, next) => {
+    const { assessmentId, participantId } = req.params;
+
+    const participantIdParsed = Number(participantId);
+    const assessmentIdParsed = Number(assessmentId);
+
+    if (
+      !Number.isInteger(participantIdParsed) ||
+      participantIdParsed < 1 ||
+      !Number.isInteger(assessmentIdParsed) ||
+      assessmentIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${participantIdParsed}" "${assessmentIdParsed}" is not a valid participant id or assessment id.`
+        )
+      );
+
+      return;
+    }
+
+    const dummyAssessmentSubmissionData = [
+      {
+        assessmentId: assessmentIdParsed,
+        principalId: participantIdParsed,
+        assessmentSubmissionStateId: 7, // graded
+        score: 10,
+        openedAt: '2023-02-09 12:00:00',
+        submittedAt: '2023-02-09 13:23:45',
+        responses: [{}],
```

Comment:
Why do we have an array of empty objects here? Why not just an empty array with nothing inside it?

Reply:
When I remove the bracket {} I get an error

Reply:
The error you get when removing the curly braces is this:

> Variable 'responses' implicitly has type 'any[]' in some locations where its type cannot be determined. ts(7034)

This means that, because you are just giving it an empty array, it doesn't know what it's supposed to be an array of. We can specify that a couple of different ways:

```typescript
const responses: Response[] = [];
```
```typescript
{
   responses: Array<Response>
}
```
Both of those above will do effectively the same thing.